malt/converters/slices.py

```diff
@@ -34,12 +34,21 @@ def _process_single_assignment(self, target, value):
     if not isinstance(target, gast.Subscript):
       return None
     s = target.slice
-    if isinstance(s, (gast.Tuple, gast.Slice)):
+    if isinstance(s, (gast.Tuple)):
       return None
 
     template = """
       target = ag__.set_item(target, key, item)
     """
+
+    if isinstance(s, (gast.Slice)):
+      template = template.replace("key", "slice(lower, upper)")
+      if s.upper is None:
+        return templates.replace(
+          template, target=target.value, lower=s.lower, upper=target.value, item=value)
+      return templates.replace(
+        template, target=target.value, lower=s.lower, upper=s.upper, item=value)
+
     return templates.replace(
         template, target=target.value, key=target.slice, item=value)
```

Comment:
Thanks, this looks approach looks nice! Could you just avoid the multiple returns structure by defining a variable for what key should be ahead of time? Then there only needs to be one replacement at the end :)